blog/2021-05-15-announcing-stryker-5-brave-new-dimension.md

```diff
@@ -0,0 +1,141 @@
+
+## 🧹 Other changes
+
+We once again chose to improve the maintainability of the Stryker code base.
+
+**Unified reporter api**
+
+The Reporter plugin API has gotten an overhaul. It is now more in line with [mutation-testing-elements](https://github.com/stryker-mutator/mutation-testing-elements/)
+
+See the [original PR](https://github.com/stryker-mutator/stryker-js/pull/2798) for more information.
+
+**Serialization**
+
+Stryker uses worker processes to run tests and check mutants for type errors. These workers receive any options via a message. Stryker used the [surrial](https://www.npmjs.com/package/surrial) package to serialize complex objects like [RegExp](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp), functions, and even classes. This hidden feature was rarely used and not a good practice in hindsight. StrykerJS 5 replaces this with plain old [`JSON.stringify`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify) and [`JSON.parse`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse). Don't worry, you get a warning if you configure something that isn't serializable.
+
+See the [original PR](https://github.com/stryker-mutator/stryker-js/pull/2877) for more information.
+
+## 💥 Breaking changes
+
+- **range:** The `range` property is no longer present on a `mutant`. Note, this is a breaking change for plugin creators only.
+- **options:** `"perTest"` is now the default value for "coverageAnalysis" when the configured test runner is not "command". Explicitly set `"coverageAnalysis": "off"` manually to opt-out of this behavior.
+- **node:** Node 10 is no longer supported. Please use Node 12 or higher.
+- **serialize:** Having a non-JSON-serializable value in your configuration won't be sent to the child process anymore. If you really need them in your test runner configuration, you should isolate those values and put them in test runner-specific config files, loaded by the test runner plugin itself, for example, jest.config.js, karma.conf.js, webpack.config.js.
+- **ignore patterns:** Stryker will no longer use a git command to determine which files belong to your project. Instead, it will rely on sane defaults. You can change this behavior by defining [`ignorePatterns`](https://stryker-mutator.io/docs/stryker-js/configuration/#ignorepatterns-string).
+- **ignore patterns:** The `files` configuration option is deprecated and will be removed in a future release. Please use [`ignorePatterns`](https://stryker-mutator.io/docs/stryker-js/configuration/#ignorepatterns-string) instead.
+
+  This:
+
+  ```json
+  {
+    "files": ["foo.js"]
+  }
+  ```
+
+  Is equivalent to:
+
+  ```json
+  {
+    "ignorePatterns": ["**", "!foo.js"]
```

Comment:
Should this be higher with the change explanation to show people how to migrate their config easier?